x/sidechannel/keeper/keeper.go

```diff
@@ -194,13 +196,14 @@ func (k Keeper) IterateValidatorsAndApplyFn(ctx sdk.Context, f func(uint64, []*a
 
 		var height uint64
 		buf := bytes.NewBuffer(heightBytes)
-		binary.Read(buf, binary.BigEndian, &height)
+
+		if err := binary.Read(buf, binary.BigEndian, &height); err != nil {
+			k.Logger(ctx).Error("Error in binary.Read")
```

Comment:
We should return after the log event.
```suggestion
			k.Logger(ctx).Error("Error in binary.Read")
			return
```